source/class/qx/ui/treevirtual/SimpleTreeDataModel.js

```diff
@@ -518,6 +518,14 @@ qx.Class.define("qx.ui.treevirtual.SimpleTreeDataModel", {
         throw new Error("Expected node object or node id");
       }
 
+      // Optimisation for removing all the nodes, node array can be reset
+      // rather than setting all the values except the root to null
+      if (nodeId === 0) {
+        this._clearSelections();
+        this._nodeArr = [qx.ui.treevirtual.MTreePrimitive._getEmptyTree()];
```

Comment:
This is a nice idea, but it is not backward compatible. Your app requires no re-rendering, but we don't know that others would appreciate losing their rerendering. I suspect that in most cases, it's wanted.

I'd propose this (untested) alternative. First, alter `clearData` as follows:

```
    clearData(bRerender = true) {
      this._checkEditing();
      this._clearSelections();
      this.setData([qx.ui.treevirtual.MTreePrimitive._getEmptyTree()], bRerender);
    },
```

Then alter `setData` signature and where it does the rerendering:

```
    setData(nodeArr, bRerender = true) {
    ...
    // Re-render the row array, if so requested
    if (bRerender) {
      this.__render();
   }

```

And of course, add documentation for `bRerender` in both functions' headers.

What do you think of this as an alternative?

Reply:
I think this will probably work fine, I'll try it out tomorrow. 

The change I proposed here doesn't prevent re-rendering itself, just the prune method didn't re-render anyway. I'm just preventing the node array getting larger every time I empty and refill the model.